README.mkd

```diff
@@ -156,6 +156,31 @@ avoid this problem.
 pseudoinstructions and whether compressed instructions are printed directly or 
 not?
 
+## Linker behaviour
+
+The RISC-V architecture supports multiple ABI conventions. Some of them are
+incompatible with one another. Two particular cases of such irreconcilable
+differencces are the floating point passing convention and whether ILP32E
+calling convention (RVE) is in use. In order to identify (ELF) object files
+built with these ABI constraints, the `e_flags` field in the ELF header contains
+a flag set to indicate the ABI in use. The details of the flag can be found in
+the [RISCV
+psABI](https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md#file-header).
+
+The linker can detect a mismatch of the ABI and indicate to the developer of the
+conflict. A mismatch should be flagged as an irrecoverable error. However,
+because the impact of the calling convention is limited to the code
+contributions of an object file, any empty object file (i.e. containing no
+sections) or any object file which does not contribute any text sections which
+are loaded at runtime shall be excluded from the requirement that the floating
```

Comment:
To be clear: this is not what LLD does and I am personally of the view it should continue to not do this. Just because BFD does something doesn't mean it's the right thing to do. There are ways to avoid this issue entirely and I am loathe to provide users with more footguns than necessary in the LLVM ecosystem, regardless of what GNU developers chose to do 5 years ago.